### PR TITLE
Add support for list copy

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98184

cc @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @soumith @desertfire

### CODE REVIEW

File: torch/_dynamo/symbolic_convert.py [hunk 1]
```diff
@@ -670,7 +670,11 @@ def LOAD_DEREF(self, inst):
         self.push(self.symbolic_locals[inst.argval])
 
     def STORE_FAST(self, inst):
-        self.symbolic_locals[inst.argval] = self.pop()
+        val = self.pop()
+        if hasattr(val, "mutable_local") and val.mutable_local:
+            if not hasattr(val.mutable_local, "source"):
+                val.mutable_local.source = LocalSource(inst.argval)
+        self.symbolic_locals[inst.argval] = val
```
Comment (jansel) [inline torch/_dynamo/symbolic_convert.py:677]: This doesn't look right.  Side effects are applied before locals are updated, and none of this should be needed for something that gets `.copy()`ed

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_dynamo/symbolic_convert.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_dynamo/symbolic_convert.py:677]: If you .copy() a ListVariable, the new ListVariable's mutable_local does not have a source. Is there a better place to bind it?

Comment (voznesenskym) [reply torch/_dynamo/symbolic_convert.py:677]: Or, we should remove the mutable_local requirement for append to work...

Comment (jansel) [reply torch/_dynamo/symbolic_convert.py:677]: Anything not an input should have `source=None`.  `val.mutable_local.source` should not be used in this example.

**Pull request closed without merging**